dask/dataframe/shuffle.py

```diff
@@ -163,15 +194,27 @@ def set_partition(
         If using the task-based shuffle, the amount of splitting each
         partition undergoes.  Increase this for fewer copies but more
         scheduler overhead.
+    empty_dataframe_detected : bool, default False
+        Whether all divisions were nan.
```

Comment:
Is this not computable from the divisions that are passed in?  I didn't quite follow the logic above.